lib/knapsack_pro/adapters/rspec_adapter.rb

```diff
@@ -54,71 +56,66 @@ def self.order_option(cli_args)
+          -> { example.id.match(/\A(.*?)(?:\[([\d\s:,]+)\])?\z/).captures.first },
+          -> { example.metadata[:file_path] },
+          -> { example.metadata[:example_group][:file_path] },
+          -> { top_level_group(example)[:file_path] },
+        ]
+          .each do |path|
+            p = path.call
+            return p if p.include?('_spec.rb')
           end
-        else
-          until example_group[:parent_example_group].nil?
-            example_group = example_group[:parent_example_group]
-          end
-        end
 
-        example_group[:file_path]
+        return ''
+      end
+
+      # private
+      def self.top_level_group(example)
+        group = example.metadata[:example_group]
+        until group[:parent_example_group].nil?
+          group = group[:parent_example_group]
+        end
+        group
       end
 
       def bind_time_tracker
-        ::RSpec.configure do |config|
-          config.prepend_before(:context) do
-            KnapsackPro.tracker.start_timer
-          end
+        ensure_no_focus!
+        log_subset_duration
+      end
 
+      def ensure_no_focus!
+        ::RSpec.configure do |config|
           config.around(:each) do |example|
-            current_test_path = KnapsackPro::Adapters::RSpecAdapter.test_path(example)
-
-            # Stop timer to update time for a previously run test example.
-            # This way we count time spent in runtime for the previous test example after around(:each) is already done.
-            # Only do that if we're in the same test file. Otherwise, `before(:all)` execution time in the current file
-            # will be applied to the previously ran test file.
-            if KnapsackPro.tracker.current_test_path&.start_with?(KnapsackPro::TestFileCleaner.clean(current_test_path))
-              KnapsackPro.tracker.stop_timer
-            end
-
-            KnapsackPro.tracker.current_test_path =
-              if KnapsackPro::Config::Env.rspec_split_by_test_examples? && KnapsackPro::Adapters::RSpecAdapter.slow_test_file?(RSpecAdapter, current_test_path)
-                example.id
-              else
-                current_test_path
-              end
+            file_path = KnapsackPro::Adapters::RSpecAdapter.file_path_for(example)
+            file_path = KnapsackPro::TestFileCleaner.clean(file_path)
 
             if example.metadata[:focus] && KnapsackPro::Adapters::RSpecAdapter.rspec_configuration.filter.rules[:focus]
-              raise "We detected a test file path #{current_test_path} with a test using the metadata `:focus` tag. RSpec might not run some tests in the Queue Mode (causing random tests skipping problem). Please remove the `:focus` tag from your codebase. See more: #{KnapsackPro::Urls::RSPEC__SKIPS_TESTS}"
+              raise "Knapsack Pro found an example tagged with :focus in #{file_path}, please remove it. See more: #{KnapsackPro::Urls::RSPEC__SKIPS_TESTS}"
```

Comment:
```suggestion
              raise "Knapsack Pro found an example tagged with :focus in #{file_path}, please remove the :focus tag. See more: #{KnapsackPro::Urls::RSPEC__SKIPS_TESTS}"
```

Be more explicit on what should be removed.

Reply:
@ArturT  That error is also triggered when you run with `--tag focus` and `fit`, so there's no `:focus` tag.

What about:

```suggestion
              raise "Knapsack Pro found an example tagged with focus in #{file_path}, please remove it. See more: #{KnapsackPro::Urls::RSPEC__SKIPS_TESTS}"
```

Reply:
Ok.